### PR TITLE
Make sure anonymous user with proper permissions can access data

### PR DESCRIPTION
**Pull request by asydorchuk**
The current implementation will throw exception when listing or accessing dashboards for the anonymous user that belongs to Public role and has proper permissions, because id and roles properties are not defined for flask AnonymousUserMixin.

### CODE REVIEW

File: caravel/views.py [hunk 1]
```diff
@@ -32,10 +32,16 @@
 log_this = models.Log.log_this
 
 
+def get_roles():
```
Comment (mistercrunch) [inline caravel/views.py]: `get_user_roles`

File: tests/core_tests.py [hunk 2]
```diff
@@ -161,6 +168,23 @@ def test_gamma(self):
         resp = self.client.get('/dashboardmodelview/list/')
         assert "List Dashboard" in resp.data.decode('utf-8')
 
+    def test_anonimous_superuser_list_dashboards(self):
```
Comment (mistercrunch) [inline tests/core_tests.py]: nit: typo on `anonymous`

File: tests/core_tests.py [hunk 3]
```diff
@@ -161,6 +168,23 @@ def test_gamma(self):
...
+        self.setup_anonimous_superuser()
+
+        resp = self.client.get('/slicemodelview/list/')
+        assert "List Slice" in resp.data.decode('utf-8')
+
+        resp = self.client.get('/dashboardmodelview/list/')
+        assert "List Dashboard" in resp.data.decode('utf-8')
+
+    def test_anonimous_superuser_access_dashboards(self):
```
Comment (mistercrunch) [inline tests/core_tests.py]: nit: typo on `anonymous`

File: caravel/views.py (hunk 1 above)
Comment (asydorchuk) [inline caravel/views.py]: Done.

File: tests/core_tests.py (hunk 2 above)
Comment (asydorchuk) [inline tests/core_tests.py]: Fixed.

File: tests/core_tests.py (hunk 3 above)
Comment (asydorchuk) [inline tests/core_tests.py]: Fixed.

**Comment by mistercrunch** [comment]
This is great overall! I like the idea  but I'm more concerned about testing public *not* having access to slices and dashboards. Maybe the tests should give access to only one dataset and confirm that only that dataset is accessible.

We may want to add a configuration flag `PUBLIC_ROLE_LIKE_GAMMA` (False by default) which would give the same access level to public as gamma does (adding missing the permissions only, not removing any), then it would be for admins to add specific datasets to the public role to allow for specific slices and dashboards.

**Comment by mistercrunch** [comment]
Let me know if you want to take on `PUBLIC_ROLE_LIKE_GAMMA` as part of this PR, we could save it for another PR if you prefer.

**Comment by asydorchuk** [comment]
I agree about testing public not having access to the specific slices or dashboards. Going to add more tests.

Regarding PUBLIC_ROLE_LIKE_GAMMA flag, let's have it as a separate pull request.

**Comment by asydorchuk** [comment]
@mistercrunch please have another look. In order to make the tests work, I needed to replace lazy_gettext with gettext method for flashing translations. The issue and behavior I was getting otherwise is described here: https://github.com/pallets/flask/issues/812

**Comment by mistercrunch** [comment]
Ok this is good! I'm planning on adding support for role definition as part of the configurations in the near future.

**Comment by asydorchuk** [comment]
Thanks! Config role definitions sounds like a great addition. We use puppet in our organization and that would simplify caravel setup on service machines. I am going to prepare a separate pull request for PUBLIC_ROLE_LIKE_GAMMA flag.

**Pull request merged**